### PR TITLE
LPD-13778 Hiding classic search portlet behind deprecation FF

### PR DESCRIPTION
**Pull request by oliv-yu**
https://liferay.atlassian.net/browse/LPD-13778 - Removing the Classic Search Portlet - Phase 1: Hide it on new instances and installations with Deprecation Feature Flag (2024.Q3)

A few backend changes were added to do the hiding/showing of search portlet on the add widget sidebar and on the configuration settings, mainly after seeing examples of `VisibilityController` and `PortletManager`.

The deprecation warning is missing the links for now, until they show up on the CDN server. The overall message is generally understandable without the link phrases (I do not think there is a setting to have a default text for liferay learn links, plus they'd be in only English), but here's what it looks like without the links:
Screenshot 2024-07-16 at 3 46 16 PM

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

**Comment by oliv-yu** [comment]
ci:report:39130181

**Comment by oliv-yu** [comment]
@thektan Pushed up the changes!

Based on Tibor's comments, there'll just be one link at the end that points to search pages, it would look like this:

Screenshot 2024-07-18 at 9 48 36 AM

Since it's not embedding links, it just reuses the language key from the feature flag. 👌

**Comment by joshuacords** [comment]
Hi @oliv-yu,
LGTM :rocket:

**Review by thektan — APPROVED** [review]
👍👍

**Comment by oliv-yu** [comment]
ci:test:search

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:forward

**Comment by oliv-yu** [comment]
ci:test:relevant

**Comment by oliv-yu** [comment]
ci:test:relevant

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/SearchPortlet.java [hunk 1]
```diff
@@ -66,6 +66,10 @@
 )
 public class SearchPortlet extends MVCPortlet {
 
+	/**
+	 * @deprecated As of Judson (7.1.x), replaced by search pages and widgets
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/SearchPortlet.java:70]: Should it be 7.4?

Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/SearchPortlet.java:70]: Hm, the ticket mentions it happens in 7.1 https://help.liferay.com/hc/en-us/articles/360017897432-Deprecated-Apps-in-7-1-What-to-Do#foundation?

**Comment by oliv-yu** [comment]
ci:report:43529791

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/SearchPortlet.java (hunk 1 above)
Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/SearchPortlet.java:70]: @gustavolimav shall mark this resolved! 😅 let me know if that's not ok, but thanks for checking as well

**Comment by oliv-yu** [comment]
ci:forward:force

**Pull request closed without merging**